### PR TITLE
Add export endpoint + split tests

### PR DESCRIPTION
**Pull request by jotak**
In the same time, refactor tests to separate perf & rest tests so that "export" is not run after perf tests.
New common file for test resources definitions

### CODE REVIEW

File: hawkular-inventory-parent/hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/InventoryService.java [hunk 1]
```diff
@@ -119,4 +119,6 @@
      * @return true is InventoryService is Up and Running
      */
     boolean isRunning();
+
+    Import buildExport();
```
Comment (lucasponce) [inline hawkular-inventory-parent/hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/InventoryService.java]: Naming types of variables is very a matter of taste.
In Alerting we finished up using plural, Import object perhaps now is weird, "Resources", "Inventory" or something that points to all the data, can be good.

Comment (lucasponce) [reply hawkular-inventory-parent/hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/InventoryService.java]: Also, I think this should be paginated.
If not a OOME is on the way.

Comment (jotak) [reply hawkular-inventory-parent/hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/InventoryService.java]: I was looking for nice names, +1 for Inventory

File: hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/handlers/InventoryHandlers.java [hunk 2]
```diff
@@ -61,6 +61,17 @@
      */
 
     @GET
+    @Path("/export")
+    @Produces(APPLICATION_JSON)
+    public Response exportInventory() {
```
Comment (lucasponce) [inline hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/handlers/InventoryHandlers.java]: Same here, pagination should be needed.
If we don't want pagination we can treated it as multipart streaming, and pagination is only in the InventoryService.
I don't know, but this is hit a OOME in the preliminar tests of QE.

File: hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java [hunk 3]
```diff
@@ -291,4 +284,53 @@ private boolean isEmpty(Collection c) {
         return c == null || c.isEmpty();
     }
 
+    @Override
+    public void buildExport(OutputStream os) throws IOException {
+        try (Writer writer = new BufferedWriter(new OutputStreamWriter(os))) {
+            writer.write("{\"types\":[\n");
```
Comment (jotak) [inline hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java]: @lucasponce do you see a better way to do it, than string-writing json? I'm stuck with StreamingOutput + paginated combination, I cannot jsonify with Jackson a yet-incomplete object. You think it's ok?

Comment (lucasponce) [reply hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java]: I know that jackson offers some streaming cappabilities, but honestly I haven't tested them yet, so only I read API and examples. Perhaps is worth to take a look and investigate this together.
I remember that in alerting the use was different as the format was not a json per se, but chunks of jsons, but on this case it seems we could rely on a library as the output is a fully json object.

File: hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java [hunk 4]
```diff
@@ -279,4 +284,53 @@ private boolean isEmpty(Collection c) {
...
+                    } else {
+                        first = false;
+                    }
+                    writer.write(JsonUtil.toJson(r) + "\n");
+                }
+                hasMore = batch.size() == MAX_RESULTS;
+                offset += MAX_RESULTS;
+            }
+            writer.write("]}\n");
+            writer.flush();
```
Comment (lucasponce) [inline hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java]: I think here the response is flushed at the end, I think it should be flushed also in intermediate steps too.
As commented, I would need to run some tests to see if I am right or I miss something in my comment.

File: hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java (hunk 3 above)
Comment (lucasponce) [reply hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java]: I remember I ran this example locally
http://www.baeldung.com/jackson-streaming-api
but I didn't investigate with detail if I can flush in the intermediate, to check that definetly we can control the buffer and avoid to have all data in memory.

Comment (jotak) [reply hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java]: Thanks, I've already seen it but it went out of my head :)

File: hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java (hunk 4 above)
Comment (jotak) [reply hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java]: I've updated with JsonGenerator, adding intermediate flush, the tests are still OK

**Comment by lucasponce** [comment]
Thx @jotak LGTM

**Pull request merged**